tests/web/test_json.py

```diff
@@ -233,6 +233,52 @@ def testGuardianResponse(client, version, buildTarget, channel, code, response):
         assert "Rule-Data-Version" in ret.headers
 
 
+@pytest.mark.usefixtures("appconfig", "guardian_db", "disable_errorhandler", "mock_autograph")
+@pytest.mark.parametrize(
+    "version,buildTarget,channel,code,response,jsonString",
+    [
+        (
+            "0.4.0.0",
+            "WINNT_x86_64",
+            "release",
+            200,
+            {"required": True, "url": "https://good.com/0.5.0.0.msi", "version": "0.5.0.0", "hashFunction": "sha512", "hashValue": "abcdef"},
+            '{"hashFunction":"sha512","hashValue":"abcdef","required":true,"url":"https://good.com/0.5.0.0.msi","version":"0.5.0.0"}'
+        ),
+        (
+            "0.6.0.0",
+            "WINNT_x86_64",
+            "release",
+            200,
+            {"required": True, "url": "https://good.com/1.0.0.0.msi", "version": "1.0.0.0", "hashFunction": "sha512", "hashValue": "mnopqr"},
+            '{"hashFunction":"sha512","hashValue":"mnopqr","required":true,"url":"https://good.com/1.0.0.0.msi","version":"1.0.0.0"}'
+        ),
+        (
+            "0.99.99.99",
+            "WINNT_x86_64",
+            "release",
+            200,
+            {"required": True, "url": "https://good.com/1.0.0.0.msi", "version": "1.0.0.0", "hashFunction": "sha512", "hashValue": "mnopqr"},
+            '{"hashFunction":"sha512","hashValue":"mnopqr","required":true,"url":"https://good.com/1.0.0.0.msi","version":"1.0.0.0"}'
+        ),
+        ("1.0.0.0", "WINNT_x86_64", "release", 404, {}, "{}"),
+        ("0.6.0.0", "Linux_x86_64", "release", 404, {}, "{}"),
+        ("0.6.0.0", "WINNT_x86_64", "beta", 404, {}, "{}"),
+        # This shouldn't match because the rule on the alpha channel contains fields not used by this type of update query.
+        ("0.6.0.0", "WINNT_x86_64", "alpha", 404, {}, "{}"),
+        ("0.6.0.0", "Darwin_x86_64", "evilrelease", 200, {}, "{}"),
+    ],
+)
+def testGuardianResponseWithCanonicalJson(client, version, buildTarget, channel, code, response, jsonString):
+    ret = client.get(f"/json/1/Guardian/{version}/{buildTarget}/{channel}/update.json")
+    assert ret.status_code == code
+    if code == 200:
+        assert ret.mimetype == "application/json"
+        assert ret.get_json() == response
```

Comment:
This (and the `response` entries it uses) are unnecessary. Comparing the raw text to the expected JSON is all we need to do here.

(Calling `get_json` is just re-parsing the raw text into a dictionary - so we're effectively checking the expected result twice.)

Reply:
Updated this as requested